components/autofill/src/autofill.udl

```diff
@@ -1,6 +1,7 @@
 namespace autofill {};
 
 dictionary NewCreditCardFields {
+    string billing_address_guid;
```

Comment:
Having the consumer specify the GUID will impact the API somewhat - that's not a problem, but worth thinking about, and maybe worth seeing how the desktop UI works.

What it means is that the address will need to be added separately, and in a different DB transaction. That's fine if UI asks for the address and card details in 2 separate forms, with 2 separate "save" buttons. However, if the UI shows just one form, it might be a problem and this API might be better defined as forcing the app to specify the address *fields*, and hide the GUID as an implementation details (ie, we would then add the address and CC in the same transaction).

We also need to define what happens when the CC is deleted - I assume the address remains.

(This is just food for thought rather than a specific suggestion - I guess the more meta point is that we really do need to start understanding what desktop does in a deeper way :)

Reply:
In the long run I absolutely do not think it makes sense for the consumer to provide an address guid in order to create a credit card. However as you specified doing anything else at this point would require that we have a better understanding of what desktop does or what the mobile team wants if that differs from desktop. Long story short I'm punting.